### PR TITLE
Add ISB External Access APIs for IntelliCode

### PR DESCRIPTION
**Pull request by tmat**
These files are copies of equivalent Razor sources.

Pythia PR switching to these APIs: https://devdiv.visualstudio.com/DevDiv/_git/Pythia/pullrequest/343326

### CODE REVIEW

**Comment by tmat** [comment]
@CyrusNajmabadi PTAL

File: src/Workspaces/Core/Portable/ExternalAccess/Pythia/Api/PythiaRemoteHostClient.cs
Comment (ryzngard) [inline src/Workspaces/Core/Portable/ExternalAccess/Pythia/Api/PythiaRemoteHostClient.cs:14]: 🥳

File: src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteHostClient.cs
Comment (ryzngard) [inline src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteHostClient.cs:11]: what warning is this and why is it disabled? (comment please)

File: src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteServiceCallbackDispatcherRegistry.cs
Comment (ryzngard) [inline src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteServiceCallbackDispatcherRegistry.cs:14]: Is this used anywhere?

**Review by ryzngard — APPROVED** [review]
LGTM, but I'm unfamiliar with Pythia related things. Seems standard for moving to OOP and External Access. Another set of eyes is preferred as well before checking this in

File: src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteServiceCallbackDispatcherRegistry.cs
Comment (tmat) [reply src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteServiceCallbackDispatcherRegistry.cs:14]: Yes, in Pythia code.

**Comment by tmat** [comment]
Merging to unblock further work. Will follow up on any feedback given after the merge in a separate PR.

**Review by CyrusNajmabadi — COMMENTED** [review]
LGTM.

**Pull request merged**
